onboard-api/src/controllers/address.controller.ts

```
@@ -66,27 +67,39 @@ export class AddressController {
         }
       }
 
-      const agreements = await this.agreementRepo.getByUserAndPool(
-        address.userId,
-        params.poolId,
-        user.email,
-        user.countryCode
-      )
-
-      const agreementLinks = agreements.map(
-        (agreement: Agreement): AgreementsStatus => {
+      // Filter profile agreements by country
+      const profileAgreements = pool.profile?.agreements
+        .filter((pa: ProfileAgreement) => {
+          return (
+            (user.countryCode === 'US' && pa.country === 'non-us') || (user.countryCode !== 'US' && pa.country === 'us')
+          )
```

Shall we throw an error here if these filters don't match?

---

That wouldn't be an error, this is just a filter to get the US agreement for US based investors and vice versa.